### PR TITLE
Local login

### PR DESCRIPTION
**Pull request by Neloop**

### CODE REVIEW

File: app/model/entity/Login.php [hunk 1]
```diff
@@ -94,6 +94,14 @@ public function passwordsMatch($password) {
     return FALSE;
   }
 
+  /**
+   * Factory method.
```
Comment (Teyras) [inline app/model/entity/Login.php]: Does this docblock convey any useful information?

File: app/model/entity/User.php [hunk 2]
```diff
@@ -316,4 +326,13 @@ public function getNameParts(): array {
       "degreesAfterName" => $this->degreesAfterName,
     ];
   }
+
+  /**
+   * True if user is have local login entity bound.
+   * @return bool
+   */
+  public function haveLocalAccounts(): bool {
```
Comment (Teyras) [inline app/model/entity/User.php]: s/have/has

File: app/model/entity/User.php [hunk 3]
```diff
@@ -316,4 +326,13 @@ public function getNameParts(): array {
       "degreesAfterName" => $this->degreesAfterName,
     ];
   }
+
+  /**
+   * True if user is have local login entity bound.
```
Comment (Teyras) [inline app/model/entity/User.php]: Returns true if the user entity is associated with a local login entity.

File: app/model/entity/Login.php (hunk 1 above)
Comment (Neloop) [reply app/model/entity/Login.php]: Just for satisfying PhpStorm...

Comment (Teyras) [reply app/model/entity/Login.php]: I can tolerate its existence, but you might want to consider turning that inspection off.

**Pull request merged**